test/providers/simulatedUser.test.ts

```diff
@@ -0,0 +1,132 @@
+import { jest } from '@jest/globals';
+import { PromptfooSimulatedUserProvider } from '../../src/providers/promptfoo';
+import { SimulatedUser } from '../../src/providers/simulatedUser';
+import type { ApiProvider, CallApiContextParams, ProviderResponse, Prompt } from '../../src/types';
+import { getNunjucksEngine } from '../../src/util/templates';
+
+jest.mock('../../src/logger', () => ({
+  debug: jest.fn(),
+  error: jest.fn(),
+}));
+
+jest.mock('../../src/util/templates', () => ({
+  getNunjucksEngine: jest.fn(),
+}));
+
+describe('SimulatedUser', () => {
+  let simulatedUser: SimulatedUser;
+  let mockOriginalProvider: ApiProvider;
+  let mockNunjucksEngine: any;
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mockOriginalProvider = {
+      id: jest.fn(() => 'mockAgent'),
+      // @ts-ignore
```

Comment:
@gru-agent remove this line. This should be `callApi: jest.fn().mockImplementation(() => async { ... }`